Backpack/Icon/Classes/BPKIcon.m

```diff
@@ -0,0 +1,236 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#import "BPKIcon.h"
+
+#import <CoreText/CoreText.h>
+
+NS_ASSUME_NONNULL_BEGIN
+NSString * const BPKIconFallbackGlyph = @"\u25A1"; // White box glyph
+
+@interface BPKIcon()
+@property(class, nonatomic, readonly) UIFont *iconFont;
+@property(class, nonatomic, readonly) NSCache<NSString *, UIImage *> *imageCache;
+@property(class, nonatomic, readonly) NSParagraphStyle *paragraphStyle;
+@property(class, nonatomic, readonly, nullable) NSDictionary *iconMapping;
+
++ (NSBundle *)iconBundle;
++ (NSString *)cacheKeyForIconNamed:(NSString *)name withColor:(UIColor *)color size:(BPKIconSize)size;
++ (NSString *)stringForUnicodeCodepoint:(nullable NSString *)codepoint;
+@end
+
+@implementation BPKIcon
+@dynamic imageCache;
+@dynamic iconFont;
+@dynamic paragraphStyle;
+@dynamic iconMapping;
+
++ (void)load {
+    NSBundle *bundle = [self iconBundle];
+
+    NSURL *url = [bundle URLForResource:@"BpkIcon" withExtension:@"ttf"];
+    NSAssert(url != nil, @"`BpkIcon.ttf` must be present in the bundle.");
+    if (url == nil) {
+        return;
+    }
+
+    NSData *fontData = [NSData dataWithContentsOfURL:url];
+
+    CGDataProviderRef provider = CGDataProviderCreateWithCFData((CFDataRef)fontData);
+    CGFontRef font = CGFontCreateWithDataProvider(provider);
+
+    if (font) {
+        CFErrorRef error = NULL;
+        if (CTFontManagerRegisterGraphicsFont(font, &error) == NO) {
+            CFStringRef errorDescription = CFErrorCopyDescription(error);
+            @throw [NSException exceptionWithName:NSInternalInconsistencyException reason:(__bridge NSString *)errorDescription userInfo:@{ NSUnderlyingErrorKey: (__bridge NSError *)error }];
```

Comment:
`errorDescription` has +1 retain count due to the copy-rule, so `__bridge` should be `__bridge_transfer`. Of course, we're the process is about the crash here, so it doesn't really matter :D

Reply:
I just stole this code from elsewhere but like you say it will crash here. I am not sure if crashing is the best option. `NSAssert` + falling back to SF and rendering http://www.fileformat.info/info/unicode/char/25a1/index.htm might be better

Reply:
Yeah, the assert + fallback is more production-friendly IMO.

Reply:
Updated